src/Home.js

```diff
@@ -1,22 +1,85 @@
-import React, { useState, useEffect } from 'react';
-import { StyleSheet, Text, View, FlatList, Pressable, Modal, TextInput, TouchableOpacity } from 'react-native';
-import { DataStore } from 'aws-amplify';
-import { Todo } from './models';
+import React, {useState, useEffect} from 'react';
+import {
+  StyleSheet,
+  Text,
+  View,
+  FlatList,
+  Pressable,
+  Modal,
+  TextInput,
+  TouchableOpacity,
+} from 'react-native';
+import {DataStore} from 'aws-amplify';
+import {Todo} from './models';
 
-export default function Home() {
+const Header = () => (
+  <View style={styles.headerContainer}>
+    <Text style={styles.headerTitle}>Todo App</Text>
+  </View>
+);
+
+const AddModal = ({modalVisible, setModalVisible}) => {
+  const [todoName, setTodoName] = useState('');
+  const [todoDescription, setTodoDescription] = useState('');
+
+  async function addTodo() {
+    await DataStore.save(
+      new Todo({
+        name: todoName,
+        description: todoDescription,
+        isComplete: false,
+      })
+    );
+    setModalVisible(false);
+    setTodoName('');
+    setTodoDescription('');
+  }
+
+  function closeModal() {
+    setModalVisible(false);
+  }
+
+  return (
+    <Modal
+      animationType="fade"
+      transparent={true}
+      visible={modalVisible}
+      onRequestClose={closeModal}
+    >
+      <View style={styles.modalOverlay}>
+        <View style={styles.addTodoContainer}>
+          <Pressable style={styles.modalDismissButton} onPress={closeModal}>
+            <Text style={styles.modalDismissText}>X</Text>
+          </Pressable>
+          <TextInput
+            style={styles.input}
+            placeholder="Name"
+            onChangeText={setTodoName}
+          />
+          <TextInput
+            style={styles.input}
+            placeholder="Description"
+            onChangeText={setTodoDescription}
+          />
+          <Pressable style={styles.buttonContainer} onPress={addTodo}>
+            <Text style={styles.buttonText}>Save</Text>
+          </Pressable>
+        </View>
+      </View>
+    </Modal>
+  );
+};
 
+const TodoList = () => {
   const [todos, setTodos] = useState([]);
-  const [todoName, setTodoName] = useState("");
-  const [todoDescription, setTodoDescription] = useState("");
-  const [modalVisible, setModalVisible] = useState(false);
 
   useEffect(() => {
     const onQuery = async () => {
       const todos_list = await DataStore.query(Todo);
       setTodos(todos_list);
     };
 
-    //load todos on first render 
+    //load todos on first render
```

Comment:
There should always be a space between the content of a comment and the `//`

Reply:
Got it.